File: HelpSoftware/Help/Help.EF/Entitys/BugReport.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class BugReport : IHelpEntity
  {
    [Key]
    public virtual int ID_BugReport { get; set; }

    public virtual User ReportUser { get; set; }

    public virtual DateTime ReportTime { get; set; }

    public virtual string Report { get; set; }

    public virtual string ReportTitle { get; set; }

    public virtual string Error { get; set; }

    public virtual bool Notified { get; set; }

    public virtual DateTime? NotfiyTime { get; set; }

    [Timestamp]
    public virtual byte[] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "BugReport";
    }

    public int GetID()
    {
      return ID_BugReport;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/BugReport.ts
import { User } from "./EF";

export class BugReport {
  ID_BugReport: Number;
  ReportUser: User;
  ReportTime: Date;
  Report: string;
  ReportTitle: string;
  Error: string;
  Notified: boolean;
  NotifyTime: Date;

  deserialize(input): BugReport {
    this.ID_BugReport = input.ID_BugReport;
    this.ReportUser = new User().deserialize(input.User);
    this.ReportTime = input.ReportTime;
    this.Report = input.Report;
    this.ReportTitle = input.ReportTitle;
    this.Error = input.Error;
    this.Notified = input.Notified;
    this.NotifyTime = input.NotifyTime;
    return this;
  }
}
